dotnet/src/SemanticKernel/CoreSkills/SemanticFunctionConstants.cs

```diff
@@ -10,11 +10,11 @@ internal static class SemanticFunctionConstants
 1. From a <goal> create a <plan> as a series of <functions>.
 2. Use only the [AVAILABLE FUNCTIONS] - do not create new functions, inputs or attribute values.
 3. Only use functions that are required for the given goal.
-4. A function has an $input and an $output.
-5. The $output from each function is automatically passed as $input to the subsequent <function>.
-6. $input does not need to be specified if it consumes the $output of the previous function.
-7. To save an $output from a <function>, to pass into a future <function>, use <function.{FunctionName} ... setContextVariable: ""$<UNIQUE_VARIABLE_KEY>""/>
-8. To save an $output from a <function>, to return as part of a plan result, use <function.{FunctionName} ... appendToResult: ""RESULT__$<UNIQUE_RESULT_KEY>""/>
+4. A function has an 'input' and an 'output'.
+5. The 'output' from each function is automatically passed as 'input' to the subsequent <function>.
+6. 'input' does not need to be specified if it consumes the 'output' of the previous function.
+7. To save an 'output' from a <function>, to pass into a future <function>, use <function.{FunctionName} ... setContextVariable: ""$<UNIQUE_VARIABLE_KEY>""/>
+8. To save an 'output' from a <function>, to return as part of a plan result, use <function.{FunctionName} ... appendToResult: ""RESULT__$<UNIQUE_RESULT_KEY>""/>
 9. Append an ""END"" XML comment at the end of the plan.
```

Comment:
>Append an ""END"" XML comment at the e [](http://example.com/codeflow?start=3&length=38)

Should there be a rule and example where the plan can't be completed?

There's actually two different cases
1. There isn't the available functions. Instead of using something else. It should indicate that it can't complete it. Either up to that step or just keep it general

2. There is the available function, but the user hasn't provided all the necessary inputs. I have found that the AI makes up an input. You could argue in this case it should be on the user to validate the inputs of the plan look correct rather than something the AI should handle

Reply:
1. Is a known area we want to improve. You're right, this is the expectation.
2. Yup, that's why create+execute are separate, so the user can validate/modify as appropriate. We'll likely be adding an object model to make it easier to manage this.